src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumClient.kt

```diff
@@ -25,31 +25,30 @@ import kotlin.time.Duration.Companion.seconds
 
 
 /** Commands */
-internal sealed interface ElectrumClientCommand {
+sealed interface ElectrumClientCommand {
     object Connected : ElectrumClientCommand
     object Disconnected : ElectrumClientCommand
-    data class ReceivedResponse(val response: Either<ElectrumResponse, JsonRPCResponse>) : ElectrumClientCommand
-    data class SendElectrumApiCall(val electrumRequest: ElectrumRequest) : ElectrumClientCommand
-    object AskForHeader : ElectrumClientCommand
+    data class AskForHeader(val callerId: UUID) : ElectrumClientCommand
+    data class ReceivedElectrumResponse(val response: Either<ElectrumResponse, JsonRPCResponse>) : ElectrumClientCommand
+    data class SendElectrumRequest(val callerId: UUID, val electrumRequest: ElectrumRequest) : ElectrumClientCommand
 }
 
 /** Actions */
-internal sealed interface ElectrumClientAction {
+sealed interface ElectrumClientAction {
+    data class SendHeader(val callerId: UUID, val height: Int, val blockHeader: BlockHeader) : ElectrumClientAction
     data class SendRequest(val request: String) : ElectrumClientAction
-    data class SendHeader(val height: Int, val blockHeader: BlockHeader) : ElectrumClientAction
-    data class SendResponse(val response: ElectrumResponse) : ElectrumClientAction
+    data class SendResponse(val callerId: UUID?, val response: ElectrumResponse) : ElectrumClientAction
```

Comment:
nit: why not have 2 different messages now:

- `SendSubscriptionResponse` without a caller id
- `SendResponse` with a caller id that cannot be `null`